celer/PN_logreg.pyx

```diff
@@ -369,15 +369,15 @@ cpdef int PN_logreg(
 
         else:
             # rescale aux to create dual point
-            norm_Xaux = dnorm_l1(
-                is_sparse, aux, X, X_data, X_indices, X_indptr,
-                notin_WS, X_mean, weights_pen, center, 0)
+            norm_Xaux = dnorm_enet(
+                is_sparse, aux, w, X, X_data, X_indices, X_indptr,
+                notin_WS, X_mean, weights_pen, center, 0, alpha, 1.0)
 
         for i in range(n_samples):
             aux[i] /= max(1, norm_Xaux / alpha)
 
-        d_obj = dual(LOGREG, n_samples, 0, &aux[0], &y[0])
-        p_obj = primal(LOGREG, alpha, Xw, y, w, weights_pen)
+        d_obj = dual(LOGREG, n_samples, alpha, 1.0, 0., 0., &aux[0], &y[0])
```

Comment:
this code is hard to understand: 1 is l1_ratio, what is the first 0 ?

Reply:
``0.`` is for ``norm2_w``
the other one is for ``norm2_y``

Reply:
I suggest to declare them as variables as you just did for l1_ratio